lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -630,7 +672,118 @@ void PatchEntryPointMutate::processShader(ShaderInputs *shaderInputs) {
   Function *entryPoint =
       addFunctionArgs(origEntryPoint, origEntryPoint->getFunctionType()->getReturnType(), argTys, inRegMask);
 
-  // Set Attributes on new function here.
+  // Set Attributes on new function.
+  setFuncAttrs(entryPoint);
+
+  // Remove original entry-point
+  origEntryPoint->eraseFromParent();
+}
+
+// =====================================================================================================================
+// Process all functions in a compute pipeline or library.
+//
+// @param shaderInputs : ShaderInputs object representing hardware-provided shader inputs
+// @param [in/out] module : Module
+void PatchEntryPointMutate::processComputeFuncs(ShaderInputs *shaderInputs, Module &module) {
+  m_shaderStage = ShaderStageCompute;
+
+  // We no longer support compute shader fixed layout required before PAL interface version 624.
+  if (m_pipelineState->getLgcContext()->getPalAbiVersion() < 624)
+    report_fatal_error("Compute shader not supported before PAL version 624");
+
+  // Determine what args need to be added on to all functions.
+  SmallVector<Type *, 8> shaderInputTys;
+  uint64_t inRegMask = generateEntryPointArgTys(shaderInputs, shaderInputTys);
```

Comment:
Please sanity-check the sizing of the SmallVectors in this function. I would expect many of them to have a typical size > 16 because of the 16 user data arguments.